src/access/wac.ts

```diff
@@ -235,3 +319,92 @@ export function getGroupAccessAll(
+  options: Partial<typeof internal_defaultFetchOptions>
+): Promise<(T & WithResourceAcl) | null> {
+  if (!hasAccessibleAcl(resource)) {
+    return null;
+  }
+  const acl = await internal_fetchAcl(resource, options);
+  const resourceWithAcl = internal_setAcl(resource, acl);
+  let resourceAcl: AclDataset;
+  if (hasResourceAcl(resourceWithAcl)) {
+    // This is the simple case, where the Resource ACL we need to update already
+    // exists.
+    resourceAcl = internal_getResourceAcl(resourceWithAcl);
+  } else if (hasFallbackAcl(resourceWithAcl)) {
+    // In this case, the Resource ACL needs to be created first, and then updated.
+    resourceAcl = createAclFromFallbackAcl(resourceWithAcl);
+  } else {
+    return null;
+  }
+
+  const resourceWithOldAcl = internal_setResourceAcl(
+    resourceWithAcl,
+    resourceAcl
+  );
+  let wacAccess: AclAccess | undefined = undefined;
+  if (hasNoUndefinedAccessModes(access)) {
+    // If all the access modes in `access` are set, no need for the previous access.
+    wacAccess = universalAccessToAcl(access, undefined);
+  } else {
+    // If some modes in `access` are undefined, they should be left unchanged.
+    // The type assumption is okay because we just set a Resource ACL for the
+    // Resource, or returned already.
+    const currentAccess = readAccess(resourceWithOldAcl, actor) as AclAccess;
+    wacAccess = universalAccessToAcl(access, currentAccess);
+  }
+
+  const updatedResourceAcl = writeAccess(resourceAcl, actor, wacAccess);
+
+  let savedAcl: AclDataset | null = null;
+  try {
+    savedAcl = await saveAclFor(resourceWithAcl, updatedResourceAcl, options);
+    return internal_setResourceAcl(resourceWithAcl, savedAcl);
+  } catch (e) {
+    return null;
+  }
+}
+
+/**
+ * Set the Access modes for a given Agent to a given Resource.
+ *
+ * Important note: if the target resource did not have a Resource ACL, and its
+ * Access was regulated by its Fallback ACL, said Fallback ACL is copied to create
+ * a new Resource ACL. This has the side effect that the next time the Fallback
+ * ACL is updated, the changes _will not impact_ the target resource.
+ *
+ * If the target Resource's Access mode cannot be determined, e.g. the user does
+ * not have Read and Write access to the target Resource's ACL, or to its
+ * fallback ACL if it does not have a Resource ACL, then `null` is returned.
+ *
+ * @param resource The Resource for which Access is being set
+ * @param agent The Agent for whom Access is being set
+ * @param access The Access being set
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns The Resource, with its ACL updated, or null if the new Access could not
+ * be set.
+ */
+export async function setAgentResourceAccess<T extends WithServerResourceInfo>(
```

Comment:
Ah and so likewise I'd call this

```suggestion
export async function setAgentAccess<T extends WithServerResourceInfo>(
```

Reply:
I agree for `setActorAccess`, which is internal and does handle both resource and default access, but I was thinking of adding https://github.com/inrupt/solid-client-js/pull/823 to the WAC API too

Reply:
Actually, that raises the question of adding `get*DefaultAccess`, which isn't part of the API yet.

Reply:
Hmm I see. I suppose we should then also consider whether we should be adding something similar to the high-level ACP API, although then the challenge is that they both have somewhat different semantics viz a viz applying to children (i.e. in WAC, it only applies to children without their own ACL, whereas in ACPs it's retroactively applied to all children).